File: cs50x/week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

//Types of dollars

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}
int x;
int count1 = 0;
int count2 = 0;
int count3 = 0;
int count4 = 0;

//Prompt the user for the change
int get_cents(void)
{
    do
    {
        x = get_int("Change owed: ");
    }
    while (x < 0);
    return x;
}
//count the quarters
int calculate_quarters(int cents)
{
    while (cents >= 25)
    {
        cents = cents - 25;
        count1++;
    }
    return count1;
}
//count the dimes
int calculate_dimes(int cents)
{
    while (cents >= 10)
    {
        cents = cents - 10;
        count2++;
    }
    return count2;
}
//count the nickels
int calculate_nickels(int cents)
{
    while (cents >= 5)
    {
        cents = cents - 5;
        count3++;
    }
    return count3;
}
//count the pennies
int calculate_pennies(int cents)
{
    while (cents >= 1)
    {
        cents = cents - 1;
        count4++;
    }
    return count4;
}

File: cs50x/week6/pset/sentimental-cash/cash.py
# TODO
import cs50


def main():
    change = positive()
    s = float_converter(change)

    quarters = 25.0
    dimes = 10.0
    nickels = 5.0
    pennies = 1.0

    count = 0
    while s >= quarters:
        count += 1
        s = s - quarters

    while s >= dimes:
        count += 1
        s = s - dimes

    while s >= nickels:
        count += 1
        s = s - nickels

    while s >= pennies:
        count += 1
        s = s - pennies

    print(f"{count}")


def positive():
    while True:
        try:
            n = float(input("Change owed: "))
            if n > 0:
                return n
        except ValueError:
            print("Not a positive num")


def float_converter(n):
    s = str(n)
    a = len(s) - 2
    b = (n % 10) / 10
    x = str(b)
    if n < 1.0:
        d = n * (10**a)
        return d
    elif len(x) > 4:
        d = n * (10**a) * 10
        return d
    else:
        return n * 100


main()
